application/src/test/java/org/opentripplanner/street/search/TemporaryVerticesContainerTest.java

```diff
@@ -0,0 +1,457 @@
+    createStreetEdge(d, c, "c -> d");
+    createStreetEdge(a, d, "a -> d");
+    g.index();
+    g.calculateConvexHull();
+  }
+
+  @Test
+  public void createFromToViaVertexRequest() {
+    var fromWithStops = new GenericLocation("From with stops", stopId, 1.0, 0.4);
+    var subject = TemporaryVerticesContainer.of(g)
+      .withFrom(fromWithStops, StreetMode.WALK)
+      .withTo(to, StreetMode.WALK)
+      .withVia(viaLocations, EnumSet.of(StreetMode.WALK))
+      .build();
+    var request = subject.createFromToViaVertexRequest();
+    assertThat(request.from()).isNotEmpty();
+    assertEquals(subject.fromVertices(), request.from());
+    assertThat(request.to()).isNotEmpty();
+    assertEquals(subject.toVertices(), request.to());
+    assertThat(request.fromStops()).isNotEmpty();
+    assertEquals(subject.fromStopVertices(), request.fromStops());
+    assertThat(request.toStops()).isEmpty();
+    assertEquals(subject.toStopVertices(), request.toStops());
+    var firstViaVertices = request.findVertices(viaLocations.get(0));
+    assertThat(firstViaVertices).isNotEmpty();
+    assertEquals(subject.visitViaLocationVertices().get(viaLocations.get(0)), firstViaVertices);
+    var thirdViaVertices = request.findVertices(viaLocations.get(2));
+    assertThat(thirdViaVertices).isNotEmpty();
+    assertEquals(subject.visitViaLocationVertices().get(viaLocations.get(2)), thirdViaVertices);
+  }
+
+  @Test
+  public void temporaryChangesRemovedOnClose() {
+    // When - the container is created
+    var subject = TemporaryVerticesContainer.of(g)
+      .withFrom(from, StreetMode.WALK)
+      .withTo(to, StreetMode.WALK)
+      .build();
+
+    // Then:
+    originAndDestinationInsertedCorrect(subject, false);
+    cleanUpAndValidate(subject);
+  }
+
+  @Test
+  public void multipleModes() {
+    try (
+      var subject = TemporaryVerticesContainer.of(g)
+        .withFrom(from, EnumSet.of(StreetMode.WALK, StreetMode.CAR_TO_PARK))
+        .withTo(to, StreetMode.WALK)
+        .build()
+    ) {
+      // When - the container is created
+
+      // Then:
+      assertThat(subject.fromVertices()).hasSize(1);
+      var fromOutgoing = subject.fromVertices().iterator().next().getOutgoing();
+      assertThat(fromOutgoing).hasSize(4);
+      assertTrue(outgoingEdgeIsTraversableWith(fromOutgoing, TraverseMode.WALK));
+      assertTrue(outgoingEdgeIsTraversableWith(fromOutgoing, TraverseMode.CAR));
+    }
+  }
+
+  @Test
+  public void temporaryChangesRemovedOnCloseWithVia() {
+    // When - the container is created
+    var subject = TemporaryVerticesContainer.of(g)
+      .withFrom(from, StreetMode.WALK)
+      .withTo(to, StreetMode.WALK)
+      .withVia(viaLocations, EnumSet.of(StreetMode.WALK))
+      .build();
+
+    // Then:
+    locationsInsertedCorrect(subject);
+    cleanUpAndValidate(subject);
+  }
+
+  @Test
+  public void locationNotFoundException() {
+    // Stops not found
+    try (
+      var container = TemporaryVerticesContainer.of(g)
```

Comment:
I think assertThrows is more appropriate here.

Reply:
assertThrows doesn't call the close() on the AutoCloseable by default, but there is a workaround https://github.com/junit-team/junit-framework/issues/2189#issuecomment-588972395. I'm not sure which solution is prettier.